### DISCUSSION TITLE
Postat status showing multiple failures

### DISCUSSION THREAD

**Discussion by MoonTechGaming** [postalserver/postal#1800] [Help with using Postal]
Hi,

After installing postal (following the official page's guidelines without any error) on my local machine using Ubuntu as a test before i have encountered the following problem:

cmd: postal status

Resulted in:
      Name                     Command               State    Ports
-------------------------------------------------------------------
postal_cron_1       /docker-entrypoint.sh post ...   Up
postal_requeuer_1   /docker-entrypoint.sh post ...   Exit 1
postal_smtp_1       /docker-entrypoint.sh post ...   Up
postal_web_1        /docker-entrypoint.sh post ...   Exit 1
postal_worker_1     /docker-entrypoint.sh post ...   Exit 1

Question:
How would one go about fixing these Exit1 codes.

System Info:
Distro: Ubuntu
Version: Ubuntu Server 20.04.3 LTS

**Comment by willpower232** [comment]
you will need to do `postal logs web` etc to see what their specific problems are

**Comment by MoonTechGaming** [comment]
My apologies, i was not aware of the logging feature in Postal, but i have fixed the Web issue, now its only the requeuer

Here are the logs.
requeuer_1  | [9] [2022-01-06T14:51:16.658] INFO -- : Running message requeuer..                            .
requeuer_1  | rake aborted!
requeuer_1  | Mysql2::Error::ConnectionError: Access denied for user 'root'@'172                            .17.0.1' (using password: YES)
requeuer_1  | /usr/local/bundle/gems/mysql2-0.5.3/lib/mysql2/client.rb:90:in `co                            nnect'
requeuer_1  | /usr/local/bundle/gems/mysql2-0.5.3/lib/mysql2/client.rb:90:in `in                            itialize'
requeuer_1  | /usr/local/bundle/gems/activerecord-5.2.6/lib/active_record/connec                            tion_adapters/mysql2_adapter.rb:22:in `new'
requeuer_1  | /usr/local/bundle/gems/activerecord-5.2.6/lib/active_record/connec                            tion_adapters/mysql2_adapter.rb:22:in `mysql2_connection'
requeuer_1  | /usr/local/bundle/gems/activerecord-5.2.6/lib/active_record/connec                            tion_adapters/abstract/connection_pool.rb:830:in `new_connection'
requeuer_1  | /usr/local/bundle/gems/activerecord-5.2.6/lib/active_record/connec                            tion_adapters/abstract/connection_pool.rb:874:in `checkout_new_connection'
requeuer_1  | /usr/local/bundle/gems/activerecord-5.2.6/lib/active_record/connec                            tion_adapters/abstract/connection_pool.rb:853:in `try_to_checkout_new_connection                            '
requeuer_1  | /usr/local/bundle/gems/activerecord-5.2.6/lib/active_record/connec                            tion_adapters/abstract/connection_pool.rb:814:in `acquire_connection'
requeuer_1  | /usr/local/bundle/gems/activerecord-5.2.6/lib/active_record/connec                            tion_adapters/abstract/connection_pool.rb:538:in `checkout'
requeuer_1  | /usr/local/bundle/gems/activerecord-5.2.6/lib/active_record/connec                            tion_adapters/abstract/connection_pool.rb:382:in `connection'
requeuer_1  | /usr/local/bundle/gems/activerecord-5.2.6/lib/active_record/connec                            tion_adapters/abstract/connection_pool.rb:1033:in `retrieve_connection'
requeuer_1  | /usr/local/bundle/gems/activerecord-5.2.6/lib/active_record/connec                            tion_handling.rb:118:in `retrieve_connection'
requeuer_1  | /usr/local/bundle/gems/activerecord-5.2.6/lib/active_record/connec                            tion_handling.rb:90:in `connection'
requeuer_1  | /usr/local/bundle/gems/activerecord-5.2.6/lib/active_record/model_                            schema.rb:466:in `load_schema!'
requeuer_1  | /usr/local/bundle/gems/activerecord-5.2.6/lib/active_record/attrib                            utes.rb:234:in `load_schema!'
requeuer_1  | /usr/local/bundle/gems/activerecord-5.2.6/lib/active_record/attrib                            ute_decorators.rb:51:in `load_schema!'
requeuer_1  | /usr/local/bundle/gems/activerecord-5.2.6/lib/active_record/model_                            schema.rb:459:in `block in load_schema'
requeuer_1  | /usr/local/bundle/gems/activerecord-5.2.6/lib/active_record/model_                            schema.rb:456:in `load_schema'
requeuer_1  | /usr/local/bundle/gems/activerecord-5.2.6/lib/active_record/model_                            schema.rb:346:in `attribute_types'
requeuer_1  | /usr/local/bundle/gems/activerecord-5.2.6/lib/active_record/model_                            schema.rb:370:in `type_for_attribute'
requeuer_1  | /usr/local/bundle/gems/activerecord-5.2.6/lib/active_record/table_                            metadata.rb:33:in `type'
requeuer_1  | /usr/local/bundle/gems/activerecord-5.2.6/lib/active_record/relati                            on/predicate_builder.rb:51:in `build'
requeuer_1  | /usr/local/bundle/gems/activerecord-5.2.6/lib/active_record/relati                            on/predicate_builder.rb:112:in `block in expand_from_hash'
requeuer_1  | /usr/local/bundle/gems/activerecord-5.2.6/lib/active_record/relati                            on/predicate_builder.rb:71:in `each'
requeuer_1  | /usr/local/bundle/gems/activerecord-5.2.6/lib/active_record/relati                            on/predicate_builder.rb:71:in `flat_map'
requeuer_1  | /usr/local/bundle/gems/activerecord-5.2.6/lib/active_record/relati                            on/predicate_builder.rb:71:in `expand_from_hash'
requeuer_1  | /usr/local/bundle/gems/activerecord-5.2.6/lib/active_record/relati                            on/predicate_builder.rb:21:in `build_from_hash'
requeuer_1  | /usr/local/bundle/gems/activerecord-5.2.6/lib/active_record/relati                            on/where_clause_factory.rb:19:in `build'
requeuer_1  | /usr/local/bundle/gems/activerecord-5.2.6/lib/active_record/relati                            on/query_methods.rb:588:in `where!'
requeuer_1  | /usr/local/bundle/gems/activerecord-5.2.6/lib/active_record/relati                            on/query_methods.rb:581:in `where'
requeuer_1  | /opt/postal/app/app/models/queued_message.rb:38:in `block in <clas                            s:QueuedMessage>'
requeuer_1  | /usr/local/bundle/gems/activerecord-5.2.6/lib/active_record/relati                            on.rb:288:in `instance_exec'
requeuer_1  | /usr/local/bundle/gems/activerecord-5.2.6/lib/active_record/relati                            on.rb:288:in `_exec_scope'
requeuer_1  | /usr/local/bundle/gems/activerecord-5.2.6/lib/active_record/scopin                            g/named.rb:186:in `block in scope'
requeuer_1  | /opt/postal/app/app/models/queued_message.rb:119:in `requeue_all'
requeuer_1  | /opt/postal/app/lib/postal/message_requeuer.rb:11:in `block in run                            '
requeuer_1  | /opt/postal/app/lib/postal/message_requeuer.rb:9:in `loop'
requeuer_1  | /opt/postal/app/lib/postal/message_requeuer.rb:9:in `run'
requeuer_1  | /opt/postal/app/lib/tasks/postal.rake:18:in `block (2 levels) in <                            top (required)>'
requeuer_1  | /opt/postal/app/bin/bundle:3:in `load'
requeuer_1  | /opt/postal/app/bin/bundle:3:in `<main>'
requeuer_1  | Tasks: TOP => postal:requeuer
requeuer_1  | (See full trace by running task with --trace)

**Reply by willpower232** [reply]
> Access denied for user 'root'@'172 .17.0.1' (using password: YES)

seems like your configuration is wrong somehow or maybe you need to try restarting Postal

**Reply by MoonTechGaming** [reply]
Hi, thanks for the quick response once again.

When i try to inspect the postal status now, all aspects are error free. However when i do a general postal logs i get the following:

cron_1      | [9] [2022-01-06T15:25:24.268] INFO -- : Starting clock for 4 events: [ every-1-minutes every-hour every-hour every-day ]
cron_1      | [9] [2022-01-06T15:25:24.270] INFO -- : Triggering 'every-1-minutes'
cron_1      | E, [2022-01-06T15:25:24.288916 #9] ERROR -- #<Bunny::Session:0x55afc79ae800 postal@127.0.0.1:5672, vhost=postal, addresses=[127.0.0.1:5672]>: Authentication with RabbitMQ failed: 403 ACCESS_REFUSED - Login was refused using authentication mechanism PLAIN. For details see the broker logfile.
cron_1      | [9] [2022-01-06T15:25:24.289] ERROR -- : Authentication with RabbitMQ failed. Please check your connection settings. Username: postal, vhost: postal, password length: 10
cron_1      | [9] [2022-01-06T15:26:24.001] INFO -- : Triggering 'every-1-minutes'
cron_1      | E, [2022-01-06T15:26:24.009438 #9] ERROR -- #<Bunny::Session:0x55afc7b41118 postal@127.0.0.1:5672, vhost=postal, addresses=[127.0.0.1:5672]>: Authentication with RabbitMQ failed: 403 ACCESS_REFUSED - Login was refused using authentication mechanism PLAIN. For details see the broker logfile.
cron_1      | [9] [2022-01-06T15:26:24.009] ERROR -- : Authentication with RabbitMQ failed. Please check your connection settings. Username: postal, vhost: postal, password length: 10
web_1       | Puma starting in single mode...
web_1       | * Version 4.3.10 (ruby 2.6.8-p205), codename: Mysterious Traveller
web_1       | * Min threads: 5, max threads: 5
web_1       | * Environment: production
web_1       | * Listening on tcp://192.168.0.242:5000
web_1       | Use Ctrl-C to stop
smtp_1      | [9] [2022-01-06T15:25:24.001] INFO -- : Listening on  :::25
requeuer_1  | [11] [2022-01-06T15:25:24.620] INFO -- : Running message requeuer...
requeuer_1  | rake aborted!
requeuer_1  | Mysql2::Error::ConnectionError: Access denied for user 'root'@'172.17.0.1' (using password: YES)
requeuer_1  | /usr/local/bundle/gems/mysql2-0.5.3/lib/mysql2/client.rb:90:in `connect'
requeuer_1  | /usr/local/bundle/gems/mysql2-0.5.3/lib/mysql2/client.rb:90:in `initialize'
requeuer_1  | /usr/local/bundle/gems/activerecord-5.2.6/lib/active_record/connection_adapters/mysql2_adapter.rb:22:in `new'
requeuer_1  | /usr/local/bundle/gems/activerecord-5.2.6/lib/active_record/connection_adapters/mysql2_adapter.rb:22:in `mysql2_connection'
requeuer_1  | /usr/local/bundle/gems/activerecord-5.2.6/lib/active_record/connection_adapters/abstract/connection_pool.rb:830:in `new_connection'
requeuer_1  | /usr/local/bundle/gems/activerecord-5.2.6/lib/active_record/connection_adapters/abstract/connection_pool.rb:874:in `checkout_new_connection'
requeuer_1  | /usr/local/bundle/gems/activerecord-5.2.6/lib/active_record/connection_adapters/abstract/connection_pool.rb:853:in `try_to_checkout_new_connection'
requeuer_1  | /usr/local/bundle/gems/activerecord-5.2.6/lib/active_record/connection_adapters/abstract/connection_pool.rb:814:in `acquire_connection'
requeuer_1  | /usr/local/bundle/gems/activerecord-5.2.6/lib/active_record/connection_adapters/abstract/connection_pool.rb:538:in `checkout'
requeuer_1  | /usr/local/bundle/gems/activerecord-5.2.6/lib/active_record/connection_adapters/abstract/connection_pool.rb:382:in `connection'
requeuer_1  | /usr/local/bundle/gems/activerecord-5.2.6/lib/active_record/connection_adapters/abstract/connection_pool.rb:1033:in `retrieve_connection'
requeuer_1  | /usr/local/bundle/gems/activerecord-5.2.6/lib/active_record/connection_handling.rb:118:in `retrieve_connection'
requeuer_1  | /usr/local/bundle/gems/activerecord-5.2.6/lib/active_record/connection_handling.rb:90:in `connection'
requeuer_1  | /usr/local/bundle/gems/activerecord-5.2.6/lib/active_record/model_schema.rb:466:in `load_schema!'
requeuer_1  | /usr/local/bundle/gems/activerecord-5.2.6/lib/active_record/attributes.rb:234:in `load_schema!'
requeuer_1  | /usr/local/bundle/gems/activerecord-5.2.6/lib/active_record/attribute_decorators.rb:51:in `load_schema!'
requeuer_1  | /usr/local/bundle/gems/activerecord-5.2.6/lib/active_record/model_schema.rb:459:in `block in load_schema'
requeuer_1  | /usr/local/bundle/gems/activerecord-5.2.6/lib/active_record/model_schema.rb:456:in `load_schema'
requeuer_1  | /usr/local/bundle/gems/activerecord-5.2.6/lib/active_record/model_schema.rb:346:in `attribute_types'
requeuer_1  | /usr/local/bundle/gems/activerecord-5.2.6/lib/active_record/model_schema.rb:370:in `type_for_attribute'
requeuer_1  | /usr/local/bundle/gems/activerecord-5.2.6/lib/active_record/table_metadata.rb:33:in `type'
requeuer_1  | /usr/local/bundle/gems/activerecord-5.2.6/lib/active_record/relation/predicate_builder.rb:51:in `build'
requeuer_1  | /usr/local/bundle/gems/activerecord-5.2.6/lib/active_record/relation/predicate_builder.rb:112:in `block in expand_from_hash'
requeuer_1  | /usr/local/bundle/gems/activerecord-5.2.6/lib/active_record/relation/predicate_builder.rb:71:in `each'
requeuer_1  | /usr/local/bundle/gems/activerecord-5.2.6/lib/active_record/relation/predicate_builder.rb:71:in `flat_map'
requeuer_1  | /usr/local/bundle/gems/activerecord-5.2.6/lib/active_record/relation/predicate_builder.rb:71:in `expand_from_hash'
requeuer_1  | /usr/local/bundle/gems/activerecord-5.2.6/lib/active_record/relation/predicate_builder.rb:21:in `build_from_hash'
requeuer_1  | /usr/local/bundle/gems/activerecord-5.2.6/lib/active_record/relation/where_clause_factory.rb:19:in `build'
requeuer_1  | /usr/local/bundle/gems/activerecord-5.2.6/lib/active_record/relation/query_methods.rb:588:in `where!'
requeuer_1  | /usr/local/bundle/gems/activerecord-5.2.6/lib/active_record/relation/query_methods.rb:581:in `where'
requeuer_1  | /opt/postal/app/app/models/queued_message.rb:38:in `block in <class:QueuedMessage>'
requeuer_1  | /usr/local/bundle/gems/activerecord-5.2.6/lib/active_record/relation.rb:288:in `instance_exec'
requeuer_1  | /usr/local/bundle/gems/activerecord-5.2.6/lib/active_record/relation.rb:288:in `_exec_scope'
requeuer_1  | /usr/local/bundle/gems/activerecord-5.2.6/lib/active_record/scoping/named.rb:186:in `block in scope'
requeuer_1  | /opt/postal/app/app/models/queued_message.rb:119:in `requeue_all'
requeuer_1  | /opt/postal/app/lib/postal/message_requeuer.rb:11:in `block in run'
requeuer_1  | /opt/postal/app/lib/postal/message_requeuer.rb:9:in `loop'
requeuer_1  | /opt/postal/app/lib/postal/message_requeuer.rb:9:in `run'
requeuer_1  | /opt/postal/app/lib/tasks/postal.rake:18:in `block (2 levels) in <top (required)>'
requeuer_1  | /opt/postal/app/bin/bundle:3:in `load'
requeuer_1  | /opt/postal/app/bin/bundle:3:in `<main>'
requeuer_1  | Tasks: TOP => postal:requeuer
requeuer_1  | (See full trace by running task with --trace)
worker_1    | [9] [2022-01-06T15:25:25.439] INFO -- : Worker running with 4 threads
worker_1    | E, [2022-01-06T15:25:25.442454 #9] ERROR -- #<Bunny::Session:0x56018973f498 postal@127.0.0.1:5672, vhost=postal, addresses=[127.0.0.1:5672]>: Authentication with RabbitMQ failed: 403 ACCESS_REFUSED - Login was refused using authentication mechanism PLAIN. For details see the broker logfile.
worker_1    | /usr/local/bundle/gems/bunny-2.14.4/lib/bunny/session.rb:1201:in `open_connection': Authentication with RabbitMQ failed. Please check your connection settings. Username: postal, vhost: postal, password length: 10 (Bunny::AuthenticationFailureError)
worker_1    |   from /usr/local/bundle/gems/bunny-2.14.4/lib/bunny/session.rb:320:in `start'
worker_1    |   from /opt/postal/app/lib/postal/rabbit_mq.rb:23:in `create_connection'
worker_1    |   from /opt/postal/app/lib/postal/rabbit_mq.rb:28:in `create_channel'
worker_1    |   from /opt/postal/app/lib/postal/worker.rb:195:in `job_channel'
worker_1    |   from /opt/postal/app/lib/postal/worker.rb:17:in `work'
worker_1    |   from script/worker.rb:3:in `<main>'


I see all my requests for the database are refused and i dont really know exactly why

**Reply by willpower232** [reply]
Unfortunately RabbitMQ and MySQL are slower to start than Postal so if they're all in Docker then its very possible for Postal to try them before they're ready.

Have a look at `WAIT_FOR_TARGETS` here https://github.com/postalserver/postal/issues/1421#issuecomment-930210579

Or maybe one of these is more appropriate https://github.com/postalserver/postal/discussions/1610
